### PR TITLE
URL->host:port for WAITER_URI if passed in that way

### PR DESCRIPTION
**Pull request by sradack**
This has wasted far too much of our time.

### CODE REVIEW

File: waiter/src/waiter/client_tools.clj [hunk 1]
```diff
@@ -71,9 +71,19 @@
       (str username "." machine-name))
     "127.0.0.1"))
 
+(defn sanitize-waiter-url
+  "Be kind to the tester, extracting the host:port of the URL if passed in as
+  a URL instead of host:port."
+  [waiter-uri]
+  (if (and waiter-uri (str/starts-with? waiter-uri "http://"))
+    (-> (URI. waiter-uri)
+        (.getAuthority))
```
Comment (dposada) [inline waiter/src/waiter/client_tools.clj]: How about

```clj
(-> waiter-uri URI. .getAuthority)
```

File: waiter/src/waiter/client_tools.clj [hunk 2]
```diff
@@ -71,9 +71,19 @@
...
+    waiter-uri))
+
 (defn retrieve-waiter-url []
   {:post [%]}
-  (let [waiter-uri (System/getenv "WAITER_URI")]
+  (let [waiter-uri (-> (System/getenv "WAITER_URI")
+                       sanitize-waiter-url)]
```
Comment (dposada) [inline waiter/src/waiter/client_tools.clj]: How about 

```clj
(-> "WAITER_URI" System/getenv sanitize-waiter-url)
```

File: waiter/src/waiter/client_tools.clj (hunk 1 above)
Comment (sradack) [reply waiter/src/waiter/client_tools.clj]: I like it.

File: waiter/src/waiter/client_tools.clj (hunk 2 above)
Comment (sradack) [reply waiter/src/waiter/client_tools.clj]: I like it.

**Pull request merged**